src/node/interfaces.cpp

```diff
@@ -854,6 +856,13 @@ class MinerImpl : public Miner
         return TestBlockValidity(state, chainman().GetParams(), chainman().ActiveChainstate(), block, chainman().ActiveChain().Tip(), /*fCheckPOW=*/false, check_merkle_root);
     }
 
+    std::unique_ptr<CBlockTemplate> createNewBlock(const CScript& scriptPubKeyIn) override
+    {
+        LOCK(::cs_main);
+        // TODO pass mempool sanely
```

Comment:
f04cf9f75054573c204245f7bb1e6813cce1fed8: I guess I should change the `BlockAssembler` constructor